pandas/tests/dtypes/test_inference.py

```diff
@@ -229,6 +233,21 @@ def __getitem__(self, item):
     assert not inference.is_list_like(NotListLike())
 
 
+def test_is_list_like_generic():
+    # GH 49649
+    # is_list_like was yielding false positives for Generic classes in python 3.11
+    T = TypeVar("T")
+
+    class MyDataFrame(DataFrame, Generic[T]):
+        ...
+
+    tstc = MyDataFrame[int]
+    tst = MyDataFrame[int]({"x": [1, 2, 3]})
+
+    assert not inference.is_list_like(tstc)
+    assert isinstance(tst, DataFrame)
```

Comment:
Might be worth also adding this test (DataFrames are list-like): `assert inference.is_list_like(tst)`

Reply:
> Might be worth also adding this test (DataFrames are list-like): `assert inference.is_list_like(tst)`

done in https://github.com/pandas-dev/pandas/pull/49736/commits/a36d1c066f0f31196d0f8318c5a98a69246e593e